src/field.rs

```diff
@@ -18,24 +18,102 @@ use crate::Ristretto255Scalar;
 #[derive(Copy, Clone, Debug)]
 pub struct FieldElement([u64; 5]);
 
+const L: FieldElement = FieldElement([
+    0x0002631a5cf5d3ed,
+    0x000dea2f79cd6581,
+    0x000000000014def9,
+    0x0000000000000000,
+    0x0000100000000000,
+]);
+const LFACTOR: u64 = 0x51da312547e1b;
+
+/// `R` = R % L where R = 2^260
+const R: FieldElement = FieldElement([
+    0x000f48bd6721e6ed,
+    0x0003bab5ac67e45a,
+    0x000fffffeb35e51b,
+    0x000fffffffffffff,
+    0x00000fffffffffff,
+]);
+
+/// `RR` = (R^2) % L where R = 2^260
+const RR: FieldElement = FieldElement([
+    0x0009d265e952d13b,
+    0x000d63c715bea69f,
+    0x0005be65cb687604,
+    0x0003dceec73d217f,
+    0x000009411b7c309a,
+]);
+
+/// u64 * u64 = u128 multiply helper
+#[inline(always)]
+fn m(x: u64, y: u64) -> u128 {
+    (x as u128) * (y as u128)
+}
+
 impl Add<FieldElement> for FieldElement {
     type Output = FieldElement;
     fn add(self, rhs: FieldElement) -> FieldElement {
-        unimplemented!();
+        let mut sum = FieldElement::zero();
+        let mask = (1u64 << 52) - 1;
+
+        // a + b
+        let mut carry: u64 = 0;
+        for i in 0..5 {
+            let total = self.0[i] + rhs.0[i] + carry;
+            carry = total >> 52;
+            sum.0[i] = total & mask;
+        }
+
+        // subtract l if the sum is >= l
+        // ?: if sum < l then it gets handled nicely?
+        sum - L
     }
 }
 
 impl Sub<FieldElement> for FieldElement {
     type Output = FieldElement;
     fn sub(self, rhs: FieldElement) -> FieldElement {
-        unimplemented!();
+        // let mut difference = FieldElement::zero();
+        // let mask = (1u64 << 52) - 1;
+
+        // // a - b
+        // let mut borrow: u64 = 0;
+        // for i in 0..5 {
+        //     let total = self.0[i].wrapping_sub(rhs.0[i] + borrow);
+        //     borrow = total >> 63;
+        //     difference.0[i] = total & mask;
+        // }
+
+        // // conditionally add l if the difference is negative
+        // unimplemented!()
```

Comment:
Let's clip this comment for now, and fix up the backing implementation later when we optimize it / make it more legible.